.github/workflows/nightly-emscripten.yml

```diff
@@ -0,0 +1,120 @@
+name: Emscripten nightly build + solc-bin push
+
+on:
+  schedule:
+    # Run once a day, at midnight
+    - cron: '0 0 * * *'
+
+jobs:
+  build-emscripten:
+    runs-on: ubuntu-latest
+    outputs:
+      nightly-version: ${{ env.NIGHTLY_VERSION }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/solidity'
+          ref: 'develop'
+
+      - name: Build soljson.js
+        run: |
+          # Note that this script will spawn and build inside a docker image (which works just fine in github actions).
+          scripts/build_emscripten.sh
+
+      - name: Get nightly version
+        id: nightly-version
+        run: |
+          last_commit_timestamp=$(git log -1 --date=iso --format=%ad HEAD)
+          last_commit_date=$(date --date="$last_commit_timestamp" --utc +%Y.%-m.%-d)
+          last_commit_hash=$(git rev-parse --short=8 HEAD)
+          solidity_version=$(scripts/get_version.sh)
+          nightly_version="v${solidity_version}-nightly.${last_commit_date}+commit.${last_commit_hash}"
+
+          echo "::set-env name=NIGHTLY_VERSION::${nightly_version}"
+
+      - name: Upload soljson.js as an artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: soljson.js
+          path: upload/soljson.js
+
+  commit-soljson:
+    runs-on: ubuntu-latest
+    needs: build-emscripten
+
+    env:
+      COMMITTER_NAME: emscripten nightly action
+      # TODO: Set a proper e-mail address
+      COMMITTER_EMAIL: solidity@example.com
+      TARGET_BRANCH: master
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      NIGHTLY_VERSION: ${{ needs.build-emscripten.outputs.nightly-version }}
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ env.TARGET_BRANCH }}
+
+      - name: Make sure that the working copy is clean
+        run: |
+          echo "::debug::$(git status)"
+          test -z "$(git status --porcelain)" || exit 1
+
+      - name: Check if the nightly already exists
+        run: |
+          version_already_exists=$(git ls-files "bin/soljson-$NIGHTLY_VERSION.js" | sed q1 || echo true && true)
+          echo "::set-env name=VERSION_ALREADY_EXISTS::${version_already_exists}"
+
+      - name: Download soljson.js artifact
+        uses: actions/download-artifact@v2
+        if: "!env.VERSION_ALREADY_EXISTS"
```

Comment:
I find this all rather hard to read - is it possible to put basically everything in a shell script? Is there an advantage in splitting this into github action steps?

Reply:
Downloading the artifact here is github action specific, so it needs to stay like this... but maybe it'd already help to merge all of the stuff below to a single ``run:`` step? But I guess @cameel may have planned to do that anyhow and it's just split so far, because that makes it quicker to see what exactly failed in the github action output.

Reply:
The advantage is that you get more useful feedback when a job fails. E.g. it shows you the values of environment variables between steps. I think that's helpful not only when writing it but would also make diagnosing future failures easier. And any more complex operations are already in scripts. The rest are just sanity checks, and simple git or file operations so basically just glue between real work.

Also, that's a habit from Ansible :) GH action syntax is **very** similar to Ansible. It's a big YAML list with steps described by dicts, each with a `name` and extra properties defining the command. The main difference being that Ansible has lots of built-in commands and you can only use one per step, which forces you to do work in small steps. In GH actions most steps just contain raw shell input so grouping it into a script is actually possible.

Another reason is that I expected that most of it would be pretty specific to GH actions and not portable. But looks like it's not that bad and with some minimal assumptions that I can document, the commands would work on any other CI.

So I can put most of it into a single script if you prefer that. I can get some of the feedback back by just adding more `echo` statements. A Bash script might also be easier to review and could sit in the `solidity` repo.

Reply:
One of the reasons I think a script would be better is that in the current format, you always have to think about adding `        if: "!env.VERSION_ALREADY_EXISTS"` instead of just using a proper if statement  with a body...

Reply:
If ``ls-files`` really doesn't need the solc-bin checkout, we could move that check to the end of the ``build_emscripten`` job above and then have the entire ``commit-soljson`` job only run if the version doesn't yet exist, i.e. have one such ``if`` at the very outer scope? Not sure that's that much better...

Reply:
Or we could just make this job fail, if the version already exists...
Actually, how is this supposed to work? Right now this will keep creating nightlies with the same commit hash every day as long as there are no commits on develop, right? Then we can assume that it's a proper error if this script runs with the version already existing and *should* just make this job fail then - then we don't need ``if``s and are notified, if this happens.
The alternative would be to only build a "new" nightly, if it's actually new, i.e. when the resulting binary is new...

Reply:
Ah right, that's basically what @chriseth is suggesting above. I'd also say, just let it fail, if it already exists, then we don't need any of this.

Reply:
Done. Moved most of the job into a script and put the whole second job in an `if`.